pxb/docker/pxb24-single-platform-run.groovy

```diff
@@ -0,0 +1,133 @@
+            name: 'BOOTSTRAP_URL',
+            trim: true)
+        choice(
+            choices: '/usr/bin/cmake',
+            description: 'path to cmake binary',
+            name: 'JOB_CMAKE')
+        choice(
+            choices: 'RelWithDebInfo\nDebug',
+            description: 'Type of build to produce',
+            name: 'CMAKE_BUILD_TYPE')
+        string(
+            defaultValue: '',
+            description: 'cmake options',
+            name: 'CMAKE_OPTS')
+        string(
+            defaultValue: '',
+            description: 'make options, like VERBOSE=1',
+            name: 'MAKE_OPTS')
+        string(
+            defaultValue: '-j 2',
+            description: './run.sh options, for options like: -j N Run tests in N parallel processes, -T seconds, -x options  Extra options to pass to xtrabackup',
+            name: 'XBTR_ARGS')
+    }
+    agent {
+        label 'micro-amazon'
+    }
+    options {
+        skipDefaultCheckout()
+        skipStagesAfterUnstable()
+        timeout(time: 6, unit: 'DAYS')
+        buildDiscarder(logRotator(numToKeepStr: '200', artifactNumToKeepStr: '200'))
+    }
+    stages {
+        stage('Check out and Build PXB') {
+            parallel {
+                stage('Build PXB24') {
+                    agent { label 'docker' }
+                    steps {
+                        git branch: 'master', url: 'https://github.com/Percona-Lab/jenkins-pipelines'
+                        echo 'Checkout PXB24 sources'
+                        sh '''
+                            # sudo is needed for better node recovery after compilation failure
+                            # if building failed on compilation stage directory will have files owned by docker user
+                            sudo git reset --hard
+                            sudo git clean -xdf
+                            sudo rm -rf sources
+                            ./pxb/docker/checkout PXB24
+                        '''
+                        echo 'Build PXB24'
+                        withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'c42456e5-c28d-4962-b32c-b75d161bff27', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                            sh '''
+                                sg docker -c "
+                                    if [ \$(docker ps -q | wc -l) -ne 0 ]; then
+                                        docker ps -q | xargs docker stop --time 1 || :
+                                    fi
+                                    ./pxb/docker/run-build-pxb24 ${DOCKER_OS}
+                                " 2>&1 | tee build.log
+
+                                if [[ -f \$(ls pxb/sources/pxb24/results/*.tar.gz | head -1) ]]; then
+                                    until aws s3 cp --no-progress --acl public-read pxb/sources/pxb24/results/*.tar.gz s3://pxc-build-cache/${BUILD_TAG}/pxb24.tar.gz; do
```

Comment:
Should we use dedicated cache for PXB instead of PXC one?

Reply:
yes, updated script.